pkg/provision/primitives/cache/cache.go

```diff
@@ -248,6 +249,91 @@ func (s *Fs) Exists(id string) (bool, error) {
 	return false, err
 }
 
+// NetworkExists exists checks if a network exists in cache already
+func (s *Fs) NetworkExists(name string, user string) (bool, error) {
+	reservations, err := s.list()
+	if err != nil {
+		return false, err
+	}
+
+	for _, r := range reservations {
+		if r.Type == primitives.NetworkReservation {
+			nr := pkg.NetResource{}
+			if err := json.Unmarshal(r.Data, &nr); err != nil {
+				return false, fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+			}
+
+			// Check if the combination of network name and user is the same
+			if nr.Name == name && r.User == user {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func (s *Fs) list() ([]*provision.Reservation, error) {
+	s.RLock()
+	defer s.RUnlock()
+
+	infos, err := ioutil.ReadDir(s.root)
+	if err != nil {
+		return nil, err
+	}
+	reservations := make([]*provision.Reservation, 0, len(infos))
+
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+
+		r, err := s.get(info.Name())
+		if err != nil {
+			return nil, fmt.Errorf("failed get reservation: %w", err)
+		}
+
+		reservations = append(reservations, r)
+	}
+	return reservations, nil
+}
+
+// incrementCounters will increment counters for all workloads
+// for network workloads it will only increment those that have a unique name
+func (s *Fs) incrementCounters(statser provision.Statser) error {
+	uniqueNetworkReservations := make(map[string]*provision.Reservation)
+
+	reservations, err := s.list()
+	if err != nil {
+		return err
+	}
+
+	for _, r := range reservations {
+		if r.Type == primitives.NetworkReservation {
+			nr := pkg.NetResource{}
+			if err := json.Unmarshal(r.Data, &nr); err != nil {
+				return fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+			}
+
+			// if the network name + user exsists in the list, we skip it.
+			// else we add it to the list
```

Comment:
Hmm, is there a way to get a unique ID from the name/user parts without doing `name + user` because this is not granted to be unique even if the user is unique !
for example a network `test` for user `user` will have the same key as the network `testu` for user `ser` !

Can u please check how the unique bridge name for the network is built ? can we reuse this code ? If it's not "clean" enough to use this code (because it belongs to a different module) you can try something like 
```go
key := hash(network) + hash(user)
```

Reply:
Check how it's done here

https://github.com/threefoldtech/zos/blob/master/pkg/provision/primitives/container.go#L187

Reply:
I think this method already have an issue and should be fixed the way I suggested above

Reply:
May be u should use this method, so once it's fixed it's fixed for all